From a3f122d59e2689c3c7b0e5508655ca9adde61619 Mon Sep 17 00:00:00 2001
From: Michael Minnick <michael.minnick@freescale.com>
Date: Fri, 28 Sep 2012 11:54:51 -0500
Subject: [PATCH 359/435] ENGR00226363 EPDC: unit test: return to RGB565 after
 test 12

Fixed test 12 to return to RGB565 format on exit. Staying in
Y8 format was casuing later tests to fail. Also fixed an
unused variable warning and incorrect test number displayed
on test failure.

Signed-off-by: Michael Minnick <michael.minnick@freescale.com>
---
 test/mxc_fb_test/mxc_epdc_fb_test.c | 20 ++++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/test/mxc_fb_test/mxc_epdc_fb_test.c b/test/mxc_fb_test/mxc_epdc_fb_test.c
index a9ce2ee..f8b9c0d 100644
--- a/test/mxc_fb_test/mxc_epdc_fb_test.c
+++ b/test/mxc_fb_test/mxc_epdc_fb_test.c
@@ -107,7 +107,6 @@ static void copy_image_to_buffer(int left, int top, int width, int height, uint
 			uint target_buf, struct fb_var_screeninfo *screen_info)
 {
 	int i;
-	__u32 *dest, *src;
 	uint *fb_ptr;
 
 	if (target_buf == BUFFER_FB)
@@ -126,11 +125,8 @@ static void copy_image_to_buffer(int left, int top, int width, int height, uint
 		return;
 	}
 
-	for (i = 0; i < height; i++) {
-		dest = fb_ptr + ((i + top) * screen_info->xres_virtual + left) * 2 / 4;
-		src = img_ptr + (i * width) * 2 /4;
+	for (i = 0; i < height; i++)
 		memcpy(fb_ptr + ((i + top) * screen_info->xres_virtual + left) * 2 / 4, img_ptr + (i * width) * 2 /4, width * 2);
-	}
 }
 
 static __u32 update_to_display(int left, int top, int width, int height, int wave_mode,
@@ -1809,6 +1805,18 @@ static int test_colormap(void)
 		WAVEFORM_MODE_GC16, TRUE, EPDC_FLAG_USE_CMAP);
 	printf("Posterized colorbar\n");
 
+	sleep(2);
+
+	printf("Change back to non-inverted RGB565\n");
+	screen_info.rotate = FB_ROTATE_UR;
+	screen_info.bits_per_pixel = 16;
+	screen_info.grayscale = 0;
+	retval = ioctl(fd_fb, FBIOPUT_VSCREENINFO, &screen_info);
+	if (retval < 0)	{
+		printf("Back to normal failed\n");
+		return TFAIL;
+	}
+
 	return retval;
 }
 
@@ -2184,7 +2192,7 @@ main(int argc, char **argv)
 	for (i = 0; i < NUM_TESTS; i++)
 		if (test_map[i])
 			if ((*testfunc_array[i])()) {
-				printf("\nError: Test #%d failed.\n", i);
+				printf("\nError: Test #%d failed.\n", i + 1);
 				goto err2;
 			}
 
-- 
1.8.0

